tests/test_temporal.py

```diff
@@ -1104,6 +1105,244 @@ async def test_toolset_without_id():
+        return result.output
+
+
+async def test_dynamic_toolset_temporal_agent_structure():
+    """Test that DynamicToolset is correctly wrapped by TemporalDynamicToolset."""
+    toolsets = dynamic_toolset_temporal_agent.toolsets
+
+    # Should have the dynamic toolset wrapped
+    assert any(isinstance(t, TemporalDynamicToolset) for t in toolsets)
+
+    # Find the TemporalDynamicToolset
+    for toolset in toolsets:
+        if isinstance(toolset, TemporalDynamicToolset):
+            # Check it has the correct id (defaults to function name)
+            assert toolset.id == 'my_dynamic_toolset'
+            # Check it has both get_tools and call_tool activities
+            activities = toolset.temporal_activities
+            assert len(activities) == 2
+            activity_names: list[str] = [
+                name
+                for a in activities
+                if (name := ActivityDefinition.must_from_callable(a).name) is not None  # pyright: ignore[reportUnknownMemberType]
+            ]
+            assert any('get_tools' in name for name in activity_names)
+            assert any('call_tool' in name for name in activity_names)
+            break
+    else:
+        pytest.fail('TemporalDynamicToolset not found in toolsets')
+
+
+async def test_dynamic_toolset_in_workflow(client: Client):
+    """Test that @agent.toolset works correctly in a Temporal workflow."""
+    async with Worker(
+        client,
+        task_queue=TASK_QUEUE,
+        workflows=[DynamicToolsetAgentWorkflow],
+        plugins=[AgentPlugin(dynamic_toolset_temporal_agent)],
+    ):
+        output = await client.execute_workflow(
+            DynamicToolsetAgentWorkflow.run,
+            args=['Call the dynamic tool with test key'],
+            id='test_dynamic_toolset_workflow',
+            task_queue=TASK_QUEUE,
+        )
+        assert output == 'Dynamic result received'
+
+
+# Test with explicit id parameter
+dynamic_toolset_agent_with_id = Agent(dynamic_toolset_function_model, name='dynamic_toolset_agent_with_id')
+
+
+@dynamic_toolset_agent_with_id.toolset(id='custom_toolset_id')
+def my_custom_id_toolset(ctx: RunContext[None]) -> FunctionToolset[None]:
+    toolset = FunctionToolset[None](id='inner')
+
+    @toolset.tool
+    def get_custom_data(key: str) -> str:
+        return f'custom_value_for_{key}'
+
+    return toolset
+
+
+dynamic_toolset_with_id_temporal_agent = TemporalAgent(
+    dynamic_toolset_agent_with_id,
+    activity_config=BASE_ACTIVITY_CONFIG,
+)
+
+
+async def test_dynamic_toolset_with_custom_id():
+    """Test that explicit id parameter is respected."""
+    toolsets = dynamic_toolset_with_id_temporal_agent.toolsets
+
+    for toolset in toolsets:
+        if isinstance(toolset, TemporalDynamicToolset):
+            assert toolset.id == 'custom_toolset_id'
+            break
+    else:
+        pytest.fail('TemporalDynamicToolset not found in toolsets')
+
+
+async def test_dynamic_toolset_get_tools_outside_workflow():
```

Comment:
We should test this by running the same agent we created earlier outside of a workflow, and checking that we still get the correct result